packages/web-components/src/components/va-pagination/va-pagination.tsx

```diff
@@ -0,0 +1,232 @@
+   * Whether or not an analytics event will be fired
+   */
+  @Prop() enableAnalytics: boolean = true;
+
+  /**
+   * The maximum number of pages to show at once
+   */
+  @Prop() maxPageListLength: number = 10;
+
+  /**
+   * The current page number
+   */
+  @Prop() page: number;
+
+  /**
+   * The total number of pages
+   */
+  @Prop() pages: number;
+
+  /**
+   * Display last page number when the page count exceeds
+   * `maxPageListLength`
+   */
+  @Prop() showLastPage: boolean;
+
+  private handlePageSelect = (page, eventID) => {
+    this.pageSelect.emit({ page });
+    if (!this.enableAnalytics) return;
+    this.componentLibraryAnalytics.emit({
+      componentName: 'va-pagination',
+      action: 'linkClick',
+      details: {
+        'event': eventID,
+        'paginate-page-number': page,
+      },
+    });
+  };
+
+  private pageNumbers = () => {
+    const {
+      maxPageListLength,
+      page: currentPage,
+      pages: totalPages,
+      showLastPage,
+    } = this;
+
+    // Make space for "... (last page number)" if not in range of the last page.
+    const showEllipsisAndLastPage =
+      showLastPage && currentPage < totalPages - maxPageListLength + 1;
+
+    const limit = showEllipsisAndLastPage
+      ? maxPageListLength - 2
+      : maxPageListLength;
+
+    let end;
+    let start;
+
+    // If there are more pages returned than the limit to show
+    // cap the upper range at limit + the page number.
+    if (totalPages > limit) {
+      start = currentPage;
+      end = limit + currentPage;
+      // treat the last pages specially
+      if (start >= totalPages - limit + 1) {
+        start = totalPages - limit + 1;
+        end = totalPages + 1;
+      }
+    } else {
+      start = 1;
+      end = totalPages + 1;
+    }
+    return Array.from({ length: end - start }, (_, i) => i + start);
+  };
+
+  private handleKeyDown = (e, pageNumber) => {
+    const keyCode = e.key;
+    if (keyCode === 'Enter' || keyCode === ' ') {
+      e.preventDefault();
+      if (!pageNumber) return;
+      this.handlePageSelect(pageNumber, 'nav-paginate-number');
+    }
+  };
+
+  render() {
+    const { ariaLabelSuffix, page, pages, maxPageListLength, showLastPage } =
+      this;
+
+    if (pages === 1) {
+      return <div />;
+    }
+
+    const renderPages = this.pageNumbers().map(pageNumber => {
+      const pageClass = classnames({
+        'button-active': page === pageNumber,
+        'button-inner': true,
+      });
+
+      return (
+        <li>
+          <button
+            aria-current={page === pageNumber ? 'true' : null}
+            aria-label={`Page ${pageNumber} ${ariaLabelSuffix}`}
+            class={pageClass}
+            onClick={() =>
+              this.handlePageSelect(pageNumber, 'nav-paginate-number')
+            }
+            onKeyDown={e => this.handleKeyDown(e, pageNumber)}
+            type="button"
+          >
+            {pageNumber}
+          </button>
+        </li>
+      );
+    });
```

Comment:
Is there a reason that the markup and possibly styles are changing?

https://github.com/department-of-veterans-affairs/component-library/blob/b346faba577fb90c0efc825e7c4268e5893f800f/packages/react-components/src/components/Pagination/Pagination.jsx#L187-L205

Reply:
Here are the only visual differences I've found:

The outline is clipped on the right side of the `50` in the React component. This is fixed in the web component.
<img width="1025" alt="Screen Shot 2022-02-08 at 20 31 47" src="https://user-images.githubusercontent.com/36863582/153104776-b33be9d3-083e-499d-ac70-398f71800a2c.png">
<img width="1022" alt="Screen Shot 2022-02-08 at 20 32 16" src="https://user-images.githubusercontent.com/36863582/153105009-35d8918e-f65c-4fbf-8e1a-b8a91926eb71.png">

The outline is smaller for the previous and next page anchors in the React component. They are bigger and consistent with the rest of the buttons in the web component.
<img width="1028" alt="Screen Shot 2022-02-08 at 20 31 58" src="https://user-images.githubusercontent.com/36863582/153104782-c99d4437-2320-4932-8eaa-95fc3a89b3d9.png">
<img width="1020" alt="Screen Shot 2022-02-08 at 20 32 23" src="https://user-images.githubusercontent.com/36863582/153105011-eb8254b3-8d83-41af-a2f9-e54c7c7fc7ba.png">

Reply:
Is this an MVP for this web component? I really dislike the current pager because as you can see in the screenshot, there is no easy way to get to page 1 &ndash; imagine being on page 25. You _have_ to click the "Prev" button until you get to the first page.

Reply:
Yes, for right now we are just focusing on replicating the React component as a web component. I agree that sounds frustrating. @rtwell would we need to go through the DSC to fix this design issue?